Sources/Fuzzilli/Compiler/Compiler.swift

```diff
@@ -1071,14 +1074,68 @@ public class JavaScriptCompiler {
     }
 
     private func mapParameters(_ parameters: [Compiler_Protobuf_Parameter], to variables: ArraySlice<Variable>) {
-        assert(parameters.count == variables.count)
-        for (param, v) in zip(parameters, variables) {
-            map(param.name, to: v)
+        var flatParameters: [String] = []
+        func extractIdentifiers(from param: Compiler_Protobuf_Parameter) {
+            switch param.parameter {
+            case .identifierParameter(let identifier):
+                flatParameters.append(identifier.name)
+            case .objectParameter(let object):
+                for property in object.parameters {
+                    extractIdentifiers(from: property.parameterValue)
+                }
+            case .arrayParameter(let array):
+                for element in array.elements {
+                    extractIdentifiers(from: element)
+                }
+            case .none:
+                break
```

Comment:
Do we ever legitimately expect `.none` for a parameter? I guess this value is autogenerated by the protobuf compiler and there's probably no way to avoid that? But maybe we should just `fatalError` in all these cases or something like that?

Reply:
No, we do not expect that. The case .none is autogenerated and switches need to be exhaustive. That is why this case is handled with just a break - it will not be reached anyway. I have included a fatal Error anyway for better clarity.